File: OrienGO-Frontend/src/app/model/dto/StudentReturn.dto.ts
import { AccountPrivacy } from '../enum/AccountPrivacy.enum';
import { EducationLevel } from '../enum/EducationLevel.enum';
import { GenderType } from '../enum/GenderType.enum';
import { MessagePermission } from '../enum/MessagePermission.enum';
import { VisibilityStatus } from '../enum/VisibilityStatus.enum';
import { LocationDTO } from './Location.dto';

export interface StudentReturnDTO {
  id: number;

  firstName: string;

  lastName: string;

  age: number;

  gender: GenderType;

  phoneNumber: string;

  email: string;

  school: string;

  fieldOfStudy: string;

  educationLevel: EducationLevel;

  location: LocationDTO;

  // Account status fields
  enabled: boolean;

  tokenExpired: boolean | null;

  lastSeen: string | null;       // ISO date string from backend
  lastLoginAt: string | null;    // ISO date string
  deletedAt: string | null;      // ISO date string

  isDeleted: boolean | null;

  suspended: boolean;

  suspensionReason: string | null;

  suspendedUntil: string | null; // ISO date string

  profileVisibility: VisibilityStatus;

  messagePermission: MessagePermission;

  accountPrivacy: AccountPrivacy;

  createdAt: string;   // ISO date string
  updatedAt: string;   // ISO date string
}

File: OrienGO-Backend/src/main/java/com/example/oriengo/model/dto/StudentReturnDTO.java
package com.example.oriengo.model.dto;

import com.example.oriengo.model.enumeration.*;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StudentReturnDTO {

    private Long id;

    private String firstName;

    private String lastName;

    private Integer age;

    private GenderType gender;

    private String phoneNumber;

    private String email;

    private String school;

    private String fieldOfStudy;

    private EducationLevel educationLevel;

    private LocationDTO location;

    // Account status fields
    private boolean enabled;

    private Boolean tokenExpired;

    private LocalDateTime lastSeen;

    private LocalDateTime lastLoginAt;

    private LocalDateTime deletedAt;

    private Boolean isDeleted;

    private boolean suspended;

    private String suspensionReason;

    private LocalDateTime suspendedUntil;

    private VisibilityStatus profileVisibility;

    private MessagePermission messagePermission;

    private AccountPrivacy accountPrivacy;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

}
